test/functional/blockchain.py

```diff
@@ -129,5 +134,18 @@ def _test_getnetworkhashps(self):
         # This should be 2 hashes every 10 minutes or 1/300
         assert abs(hashes_per_second * 300 - 1) < 0.0001
 
+    def _test_stopatheight(self):
+        assert_equal(self.nodes[0].getblockcount(), 200)
+        self.nodes[0].generate(6)
+        assert_equal(self.nodes[0].getblockcount(), 206)
+        self.log.debug('Node should not stop at this height')
+        assert_raises(subprocess.TimeoutExpired, lambda: bitcoind_processes[0].wait(timeout=3))
+        self.nodes[0].generate(1)
```

Comment:
I believe this call to generate might fail -- isn't it possible the node's shutdown will interrupt the RPC response?  That looks to me like what happened here:
https://travis-ci.org/bitcoin/bitcoin/jobs/245917226#L2275

I think we should wrap the generate(1) in a try...except and allow for the RPC to fail due to connection refused, etc.